File: backend/models/story.go
package models

import "time"

type Story struct {
	Id        int64     `json:"id"`
	Author    Profile   `json:"author"`
	Media     Media     `json:"media"`
	CreatedAt time.Time `json:"createdAt"`
}

type CreateStoryReq struct {
	Author Profile `json:"author"`
	Media  Media   `json:"media"`
}

type StoriesRes struct {
	Author  Profile `json:"author"`
	Stories []Story `json:"stories"`
}

File: android/app/src/main/java/bagus2x/sosmed/domain/model/Story.kt
package bagus2x.sosmed.domain.model

import java.time.LocalDateTime

data class Story(
    val id: Long,
    val author: Author,
    val medias: List<Media>,
    val createdAt: LocalDateTime,
) {

    val unseen: Boolean
        get() = medias.any { media ->
            when (media) {
                is Media.Image -> media.unseen
                is Media.Video -> media.unseen
            }
        }

    data class Author(
        val userId: Long,
        val photo: String?,
        val username: String,
    )
}
